### PR TITLE
Improvement/106 code docs generation

### LINKED ISSUE #106: Source code doc generation
**Description**
Doxygen + Breathe + Sphinx (and maybe Exhale):
- Source code API generation
- [C++ + CMake + Doxygen + Breathe + Sphinx](https://devblogs.microsoft.com/cppblog/clear-functional-c-documentation-with-sphinx-breathe-doxygen-cmake/)
- [Breathe official page](https://breathe.readthedocs.io/en/latest/index.html)
  - Note that this page uses the **Read-the-Docs** theme
- [C++ documentation with Doxygen/CMake/Sphinx/Breathe](https://medium.com/practical-coding/c-documentation-with-doxygen-cmake-sphinx-breathe-for-those-of-use-who-are-totally-lost-part-2-21f4fb1abd9f)
- With Exhale:
  - [Example](https://rgoswami.me/posts/doc-cpp-dox-sph-exhale/)
  - [Explanation](https://marian-nmt.github.io/docs/api/doc_guide.html#user-documentation-with-sphinx)

MKDocs Approach:
- [MkDocs](https://www.mkdocs.org/getting-started/)
- DFR-wide documentation site?
- Add a custom DFR theme?? Taggable light & dark modes?
- [Collection repositories of MkDoc tools](https://github.com/mkdocs/catalog#-api-documentation-building)

**Sub-tasks**
- [ ] #

**Definition of Done**
* Tested
* Code reviewed

### PR DESCRIPTION
**Pull request by crsz20**
### Summary
Documentation that is built from scanning the firmware source code.
- Comments are made using `///`.
- `.rst` files are "restructured text" files. These are kind of like how `.md` Markdown files work, but with its own rules. We will not be using this extensively as I intend to write most of our documentation in our wiki.
- Steps to reproduce is documented in our [Source Code Docs](https://github.com/DallasFormulaRacing/DataAcquisition2.0/wiki/Source-Code-Docs) page.

### Remarks
Feel free to leave suggestions on anything. Color palette, content to add, more descriptive comments, etc.

Building similar docs for ST's HAL library did not seem to work. Their `.c` files had similar doxygen-style comments, but doxygen fails to parse their code. We will have to keep relying solely on their massive PDF for their API documentation.

### CODE REVIEW

File: Project/DAQ_System/Program/Src/Sensor/LinearPotentiometer/sls1322.hpp [hunk 1]
```diff
@@ -26,25 +26,36 @@ namespace sensor {
...
+        /// Resolution for a 12-bit ADC, 2^(12 bits) - 1
+        static constexpr uint16_t kMaxResolution = 0x0FFF;
         static constexpr uint8_t kMaxLengthInches = 3;
         static constexpr float kMaxLengthMillimeters = 25.4f * kMaxLengthInches;
+
+        /// The ADC peripheral from ST's HAL library.
+        ADC_HandleTypeDef& adc_;
 };
 
 }
```
Comment (crsz20) [inline Project/DAQ_System/Program/Src/Sensor/LinearPotentiometer/sls1322.hpp:61]: Page after rendering:

image

File: Project/Docs/_static/breathe.css [hunk 2]
```diff
@@ -0,0 +1,113 @@
+/* -- breathe specific styles ----------------------------------------------- */
```
Comment (crsz20) [inline Project/Docs/_static/breathe.css:1]: This file is simply for customizing the color scheme.

File: Project/Docs/conf.py [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+            "url": "https://github.com/DallasFormulaRacing/DataAcquisition2.0",
+            "html": """
+                <svg stroke="currentColor" fill="currentColor" stroke-width="0" viewBox="0 0 16 16">
+                    <path fill-rule="evenodd" d="M8 0C3.58 0 0 3.58 0 8c0 3.54 2.29 6.53 5.47 7.59.4.07.55-.17.55-.38 0-.19-.01-.82-.01-1.49-2.01.37-2.53-.49-2.69-.94-.09-.23-.48-.94-.82-1.13-.28-.15-.68-.52-.01-.53.63-.01 1.08.58 1.23.82.72 1.21 1.87.87 2.33.66.07-.52.28-.87.51-1.07-1.78-.2-3.64-.89-3.64-3.95 0-.87.31-1.59.82-2.15-.08-.2-.36-1.02.08-2.12 0 0 .67-.21 2.2.82.64-.18 1.32-.27 2-.27.68 0 1.36.09 2 .27 1.53-1.04 2.2-.82 2.2-.82.44 1.1.16 1.92.08 2.12.51.56.82 1.27.82 2.15 0 3.07-1.87 3.75-3.65 3.95.29.25.54.73.54 1.48 0 1.07-.01 1.93-.01 2.2 0 .21.15.46.55.38A8.013 8.013 0 0 0 16 8c0-4.42-3.58-8-8-8z"></path>
+                </svg>
+            """,
+            "class": "",
+        },
+    ],
+}
```
Comment (crsz20) [inline Project/Docs/conf.py:101]: Neat footnote icons 🤩

image

File: Project/Docs/index.rst [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+- Low-level code.
+- Consists of drivers. That is, libraries to support sensors and external devices.
+- Product-specific code.
+
+Platform Layer
+^^^^^^^^^^^^^^
+
+- Low-level code.
+- Access to microcontroller peripherals.
+- Platform-specific code. Such as, making use of I2C on an STM32 with the HAL library vs. on a TI microcontroller.
```
Comment (crsz20) [inline Project/Docs/index.rst:59]: After rendering:

image

**Review by WilliamLim781 — APPROVED** [review]
looking pretty nice! good job!

**Pull request merged**